src/precompile.jl

```diff
@@ -3,5 +3,41 @@ function _precompile_()
     precompile(Tuple{typeof(Parsers.parse), Type{Int64}, String})
     precompile(Tuple{typeof(Parsers.parse), Type{Float64}, String})
     precompile(Tuple{typeof(Parsers.parse), Type{Date}, String})
+
+    options = Parsers.Options()
+    pos = 1
+    val = "a"
+    len = length(val)
+    for T in (Char, String), buf in (codeunits(val), Vector(codeunits(val)))
+        Parsers.xparse(T, buf, pos, len, options)
+        Parsers.xparse(T, buf, pos, len, options, Any)
+    end
+
+    for T in (Char, Symbol), buf in (val, Vector(codeunits(val)))
+        Parsers.xparse2(T, buf, pos, len, options)
+        Parsers.xparse2(T, buf, pos, len, options, T)
+        Parsers.xparse2(T, buf, pos, len, options, Any)
+    end
+
+    val = "123"
+    len = length(val)
+    for T in (Int8, Int16, Int32, Int64, Float16, Float32, Float64, BigFloat, Dates.Date, Dates.DateTime, Dates.Time, Bool), 
+        buf in (codeunits(val), Vector(codeunits(val)))
+        Parsers.xparse(T, buf, pos, len, options)
+        Parsers.xparse(T, buf, pos, len, options, T)
+        Parsers.xparse(T, buf, pos, len, options, Any)
+    end
+    for T in (Int8, Int16, Int32, Int64, Float16, Float32, Float64, BigFloat, Dates.Date, Dates.DateTime, Dates.Time, Bool), 
+        buf in (val, SubString(val, 1:3), Vector(codeunits(val)), view(Vector(codeunits(val)), 1:3))
```

Comment:
Yeah, I wondered about that too. The problem is this affects so many packages, it's hard to asses what the required methods are so I did (nearly) everything.

`Bool`, `String`, `Symbol`, `Time` are arbitrarily not included in some precompilations because they errored, probaly because they probably need different inputs and I don't know all the details of the package.

It does save 12 seconds in CSV.jl and 5 in Blink.jl TTFX, probably more in the other 2198 packages. So the 30s precompilation is likely worth it. 

But if anyone has an idea how to reduce this list in an informed way that would help.

Reply:
Here would be my preference:
  * precompile `String`, `Int32`, `Int64`, `Float64`, `Bool`, `Date`, and `DateTime` for `xparse` and `tryparse` (which will include `xparse2`)
  
  I think it would be rare for people to actually need `Char`, `Symbol`, small integer types, `Float16`. Precompiling `parse`, `tryparse`, and `xparse2` should all be redundant, we should only need one most likely. 
  
  We could probably leave in `BigFloat` since it has a number of complicated code paths that would be good to exercise.
  
  Hopefully that trims down the precompilation cost quite a bit? 30seconds seems exorbitant, like, that's how long DataFrames precompilation takes??

Reply:
Thanks, these types and methods seem to work fine for Blink and CSV, and the precompile time is less than half. 

To clarify did you mean not to precompile `Float32`? as its not in your list of rarely used types.

Reply:
Yeah, I'd say leave that out. I don't think it's as common to parse Float32 vs. Float64. Glad that brings the precompile time down so much.